developer_docs/foreman_dev_setup.asciidoc

```diff
@@ -0,0 +1,115 @@
+
+[[prerequisites]]
+== Prerequisites
+Foreman will run with the following requirements (aside from rubygem dependencies):
+
+* Ruby v2.7.x
+* NodeJS v12 or v14
+* NPM v6
+* PostgreSQL v13
+
+It’s recommended to install https://github.com/rbenv/rbenv[rbenv] to manage ruby, and https://github.com/nvm-sh/nvm[nvm] to manage node and npm on your system.
+Before installing PostgreSQL, check first that you don't already have it installed (by running `rpm -q postgresql`).
+
+You will want to make sure that you have postgresql-devel installed so the database gems can install properly.
+Also, you would also need gcc, ruby-devel, libxml-devel, libxslt-devel and libvirt-devel packages.
+
+For RHEL7 or clones, you can issue the following commands to install all required packages:
+[source, bash]
+....
+yum groupinstall "Development Tools" "Development Libraries"
+yum -y install gcc-c++ git ruby ruby-devel rubygems \
+    libvirt-devel postgresql-devel openssl-devel \
+    libxml2-devel libxslt-devel zlib-devel \
+    readline-devel systemd-devel tar libcurl-devel
+....
+
+[[Setup]]
+== Setting the repository
+[source, bash]
+....
+# On Github create your own fork
+git clone https://github.com/{GITHUB_USER}/foreman.git -b develop
+cd foreman
+cp config/settings.yaml.example config/settings.yaml
+cp config/database.yml.example config/database.yml
+git remote add upstream git@github.com:theforeman/foreman.git
+gem install bundler
+bundle install
+npm install
+....
+
+[[Database]]
+== Setting the database
+It is important that config/database.yml is set to use the correct database in the “development” block.
+Rails (and subsequently Foreman) will use these connection settings under “development” to manage the database it uses and setup the necessary schema.
+
+If the default version of the database.yml file isn't working for you, try editing it with the following:
+[source, ruby]
+....
+development:
+ adapter: postgresql
+ host: localhost
+ username: postgres
+ password: password
+ pool: 5
+ database: foreman_development
+
+test:
+ adapter: postgresql
+ host: localhost
+ username: postgres
+ password: password
+ pool: 5
+ database: foreman_test
+....
+
+Set up database schema, precompile assets and locales:
+[source, ruby]
+....
+bundle exec rake db:create # if not already created
+bundle exec rake db:migrate
+bundle exec rake db:seed assets:precompile locale:pack webpack:compile
+....
+
+The db:seed step will print out the default admin password, record this in order to log in later.
+
+[[Running]]
+== Running Foreman
+In order to run Foreman you can use the following command inside your git repository:
+[source, ruby]
+....
+bundle exec foreman start
```

Comment:
yeah, that's really good!
there's also a way to configure an options file for things like the `NOTIFICATIONS_POLLING`
cc @MariaAga

Reply:
`NOTIFICATIONS_POLLING` and `REDUX_LOGGER` are options that can be set in an `.env` file as well as in the cli call, maybe worth mentioning.
 `REDUX_LOGGER` has a boolean value and controls if the foreman will print each redux call in the web console

Reply:
as discussed offline,
it would be better to execute `bundle exec foreman start webpack` 
which will trigger the webpack script in Procfile: https://github.com/theforeman/foreman/blob/develop/Procfile#L5
@MariaAga how does your `.env` file looks like? 

just like that?
```env
NOTIFICATIONS_POLLING = 99999
REDUX_LOGGER = false
```

Reply:
In the core only setup, yes.
With forklift I also have: (but I think that might be for a more in-depth guide)
```
BIND=0.0.0.0
PORT=3000
RAILS_STARTUP='puma -w 2 -p $PORT --preload'
WEBPACK_OPTS='--https --key /etc/pki/katello/private/katello-apache.key --cert /etc/pki/katello/certs/katello-apache.crt --cacert /etc/pki/katello/certs/katello-default-ca.crt --host 0.0.0.0 --public centos7-luna-devel.fedora.example.com'
REDUX_LOGGER=false
NOTIFICATIONS_POLLING=1000000
```

Reply:
@Ron-Lavi, do you need me to change something related to this?

Reply:
Resolving this thread, thanks for adding the `.env` file explanation in the lower section 👍🏼